zeppelin-server/src/main/java/org/apache/zeppelin/rest/InterpreterRestApi.java

```diff
@@ -115,9 +116,10 @@ public Response updateSetting(String message, @PathParam("settingId") String set
           UpdateInterpreterSettingRequest.class);
       interpreterFactory.setPropertyAndRestart(settingId, p.getOption(), p.getProperties());
     } catch (InterpreterException e) {
-      return new JsonResponse(Status.NOT_FOUND, e.getMessage(), e).build();
+      return new JsonResponse(Status.NOT_FOUND, e.getMessage(), settingId).build();
```

Comment:
I think it's still useful to log the exception. setPropertyAndRestart() and restart() may return InterpreterException not oriented from setting id not found. The exception possibly can come from lose(), destroy() inside of setPropertyAndRestart() or start().